lib/blueprinter/helpers/base_helpers.rb

```diff
@@ -6,6 +6,8 @@ def self.included(base)
 
     module SingletonMethods
       include TypeHelpers
+      include AssociationHelpers
+      @@should_preload_associations = false
```

Comment:
Two things: 
1. Instead of `should_preload_associations`, it should just be `preload_associations`. Otherwise it seems unsure and somewhat like a test file.
2. This should be in configurations.rb https://github.com/procore/blueprinter/blob/60b8f94616e18aa0346fa3c697b9e30c66c4ec8c/lib/blueprinter/configuration.rb#L3-L18 so that users can configure their options like everything else.

Reply:
I see what you are saying here @philipqnguyen . One thing to note is that the `preload_associations` variable you are referencing is not meant to be a permanent field. It would only exist until we are done testing this feature. After that time, blueprinter will try to eager load associations that would otherwise not be eager loaded. My main concerns are: 

- I will need to introduce more code into [task_blueprint](https://github.com/procore/procore/blob/7862057bfaf4faad88f1c9345f27e64b989ee788/app/blueprints/schedule/task_blueprint.rb#L46) in the monolith. 
- Developers in the monolith may start using this option which will could result in blueprinter continuing to user eager loading even if we turn the flag off. This would be a problem if a bug arises and we want to roll back the eager loading changes in blueprinter. We will not be able to do so without a redeploy

Reply:
Blueprinter is a public gem and changes to Blueprinter should be ready for use by the greater Ruby community. 
If the goal in this PR is just to test this feature, I suggest 2 things:
a) Either have the monolith's Gemfile reference this branch
b) Or directly override the monolith's Blueprinter with these changes (basically a monkey patch).

When you have completed the test in the monolith, then we can introduce the change into Blueprinter itself.

Reply:
Since we want to scope the changes to only affect the monolith, I like the idea of pointing the monolith's Gemfile to this branch. I would also like to be able to toggle this feature with a feature flag, as opposed to a config that would allow developers to use the feature in other places as we test. The idea being that we would only scope the change to a narrow part of the code (extended view of tasks controller).

I would propose:
1. Pointing the monolith to this local branch via the Gemfile as you suggested.
2. Using the feature flag as is currently implemented on this branch

This approach would decrease the likelihood of needing a redeploy in case something goes wrong because we would be able to toggle the feature flag off.

Please let me know what you think. Thank you